hack/pre-release.sh

```diff
@@ -0,0 +1,72 @@
+# this script updates the WMCO and OCP versions in the Makefile, base CSV, bundle and Dockerfiles.
+
+# usage: 
+# ./pre-release (x, y or z) 
+# to update a z stream, use ./pre-release.sh z
+# for other streams, use the corresponding x y or z argument.
+
+update_WMCO_version(){
+  xyz_stream=$1
+  cluster_service_version_yaml=bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml
+  
+  if [[ $xyz_stream == "x" ]]; then
+    stream_var=1
+  fi 
+  if [[ $xyz_stream == "y" ]]; then
+    stream_var=2
+  fi 
+  if [[ $xyz_stream == "z" ]]; then
+    stream_var=3
+  fi
+  
+  version=$(sed -n -e 's/^.*WMCO_VERSION ?= //p' Makefile)
+  updated_version=$(echo $version | 
+    awk -F. -v OFS=. -v stream_var="$stream_var" '{$stream_var += 1 ; print}')
+  last_version=$(grep "olm.skipRange" $cluster_service_version_yaml | 
+    grep -oP ">=\K([0-9]+\.[0-9]+\.[0-9]+)\s*<" | sed "s/\s*<//" | 
+    awk -F. -v OFS=. -v stream_var="$stream_var" '{$stream_var += 1 ; print}')
+
+  echo "WMCO version $version detected from the Makefile"
+
+  echo "Updating the Makefile to version $updated_version"
+  sed -i "s/$version/$updated_version/g" Makefile
+ 
+  echo "Updating the cluster service version yaml to $updated_version"
+  sed -i "s/$version/$updated_version/g" $cluster_service_version_yaml
+  
+  echo "Updating previous version to $version"
+  sed -i "s/$last_version/$version/g" $cluster_service_version_yaml
+}
+
+update_OCP_version() {
+  xy_stream=$1
+  cluster_service_version_yaml=config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml
+  
+  if [[ $xy_stream = "x" ]]; then 
+    stream_var=2
+  else
+    exit 
+  fi 
+
+  version=$(grep -o -E 'OCP \b([0-9]\.[0-9]{2})\b' $cluster_service_version_yaml | 
+    sed -E 's/OCP ([0-9]\.[0-9]{2})/\1/')
```

Comment:
Is there a jq equivalent that you can use?

Reply:
This has not been answered.

Reply:
This continues to not be answered.

Reply:
I'm getting this from a comment inside the clusterserviceversion.yaml (the description [here](https://github.com/openshift/windows-machine-config-operator/blob/cc553e29d3bc16af106392e013e77c3ac4c5428d/config/manifests/bases/windows-machine-config-operator.clusterserviceversion.yaml#L37)) 
If I were to use JQ I would end up searching the comment anyway. 
Where I'm getting this information is arbitrary though, and I could get it from somewhere else, but I figured that getting it from the doc comment would be a reliable source.

Reply:
I was not asking you to use `jq` but asking if there is a similar utility for processing yaml files. Given the amount of search and replacing with regexes you are doing here, I am wondering if there is a more elegant way of making these changes. Saying you need an utility as a pre-req is acceptable given it will make the code easier to read, understand and maintain. The way it stands, it looks pretty fragile and prone to breakage.

PS: This is a general comment and just specific to this line.

Moreover getting the version from a comment is not advisable. What if we decide to change that comment. Lets figure out a better spot to get the version from. It needn't be from the CSV but elsewhere.

Reply:
Please add a comment saying what exactly you are looking for with your search statements. It is in no way obvious that you were picking the version from `* OCP 4.14 cluster running on Azure, AWS, GCP or vSphere configured with hybrid OVN Kubernetes networking`. Do this for all the lines.

Reply:
how about from the gitmodules [here](https://github.com/openshift/windows-machine-config-operator/blob/cc553e29d3bc16af106392e013e77c3ac4c5428d/.gitmodules#L4)